src/TitleDetails/AddPieceModal/AddPieceModal.js

```diff
@@ -179,6 +184,30 @@ const AddPieceModal = ({
     onSave();
   }, [change, onSave]);
 
+  const onUnreceivePiece = useCallback(() => {
+    const currentPiece = {
+      ...formValues,
+      checked: true,
+    };
+
+    return unreceive([currentPiece])
```

Comment:
It seems to me that all this stuff should be handled on <TitleDetailsEditContainer> level and modal should only get a CB as a prop